pkg/graph/resolvers/it_gov_task_statuses.go

```diff
@@ -57,6 +58,24 @@ func GrbMeetingStatus(intake *models.SystemIntake) models.ITGovGRBStatus {
 }
 
 // DecisionAndNextStepsStatus calculates the ITGovDecisionStatus for the Decisions section for the system intake task list for the requester view
-func DecisionAndNextStepsStatus(intake *models.SystemIntake) models.ITGovDecisionStatus {
-	return models.ITGDSCantStart
+func DecisionAndNextStepsStatus(intake *models.SystemIntake) (models.ITGovDecisionStatus, error) {
+	if !(intake.Step == models.SystemIntakeStepGRBMEETING || intake.Step == models.SystemIntakeStepDECISION) { // If not in either of these steps, we can't issue a decision yet. Note, this does not check that the step is a valid enum
+		return models.ITGDSCantStart, nil
+	}
+	if intake.Step == models.SystemIntakeStepGRBMEETING {
+		if intake.GRBDate == nil {
+			return models.ITGDSCantStart, nil
+		}
+		if intake.GRBDate.After(time.Now()) { // Meeting has not happened
+			return models.ITGDSCantStart, nil
+		}
+		// Meeting has  happened, intake is waiting on a decision
+		return models.ITGDSInReview, nil
+	}
+	// if the step is decision state, the step has to be completed, and a decision has to have been made
+	if intake.DecisionState == models.SIDSNoDecision {
+		return "", apperrors.NewInvalidEnumError(fmt.Errorf("intake has an invalid value for its decision state. a decision must be made in order to be in the decision state"), intake.RequestFormState, "SystemIntakeDecisionState")
+	}
```

Comment:
The comment says `if the step is decision state`, but that only seems to be enforced by the order of the previous early returns. Is it worth doing an explicit check for that case?
```suggestion
	// if the step is decision state, the step has to be completed, and a decision has to have been made
	if intake.DecisionState == models.SIDSNoDecision && intake.Step == models.SystemIntakeStepDECISION {
		return "", apperrors.NewInvalidEnumError(fmt.Errorf("intake has an invalid value for its decision state. a decision must be made in order to be in the decision state"), intake.RequestFormState, "SystemIntakeDecisionState")
	}
```

Reply:
I guess my main thoughts on this are:

- It'd be nice if we explicitly handle the cases we _know_ about in the main body of the function
- It'd be nice if we remove, add, or change  steps/states that tests and functions like this stop working so they have to be touched to ensure we're covering new cases (i.e., handle all our cases with a `return`, and return an `error` at the end if we didn't catch any of our expected cases?)

Reply:
Thanks @ClayBenson94 , I refactored this to use a switch statement, so if new steps are added, they are handled.  The other two implementations don't do this, but look specifically to see if they are in the first step, and if not return that the state is complete.

 I think those work, but we probably should handle each state explicitly for the other tickets.